trieste/ask_tell_optimization.py

```diff
@@ -260,6 +260,7 @@ def ask(self) -> TensorType:
         #   which, when called, returns state and points
         # so code below is needed to cater for both cases
 
+        # TODO: handle step number for logging?
```

Comment:
With configurable step number now we can just off-load that to the user. Some explanation in the docstring of the optimizer would suffice.